docs/how-to/use_different_stage_games.rst

```diff
@@ -47,3 +47,28 @@ The default Prisoner's dilemma has different results::
     >>> results = tournament.play()
     >>> results.ranked_names
     ['Defector', 'Tit For Tat', 'Cooperator']
+
+Asymmetric games can also be implemented via the AsymmetricGame class 
+with two Numpy arrays for payoff matrices::
+
+    >>> import numpy as np
+    >>> A = np.array([[3, 1], [1, 3]])
+    >>> B = np.array([[1, 3], [2, 1]])
+    >>> asymmetric_game = axl.AsymmetricGame(A, B)
+    >>> asymmetric_game  # doctest: +NORMALIZE_WHITESPACE
+    Axelrod game with matrices: (array([[3, 1],
+                                        [1, 3]]),
+                                 array([[1, 3],
+                                        [2, 1]]))
+
+Asymmetric games can also be different sizes, such as Rock Paper Scissors::
```

Comment:
Yup great. :)

Reply:
Perhaps we should hold off on mentioning non 2x2 games, other stuff may break like the mapping of "2" to a non-existent action. Maybe in a follow-up PR a full rock-paper-scissors example could be added, with a few actual strategies (all R, all P, all R, random, etc.).

Reply:
> Maybe in a follow-up PR a full rock-paper-scissors example could be added, with a few actual strategies (all R, all P, all R, random, etc.).

Yeah, good call.

Reply:
I'd prefer to leave it in with a health warning (that it's experimental and some things break) because as it stands, creating larger games, strategies for those larger games, and matches and tournaments between those strategies also work. What doesn't work (such as transformers) are algorithms and objects particularly created for the study of the Iterated Prisoner's Dilemma, which isn't a fault of either the algorithm, strategy or game, just inappropriate application of said algorithms to those strategies and games.

The wider solution (which I'd recommend doing in a follow-up PR) would be to classify strategies by what game they are created for, and raise a warning/error if an inappropriate algorithm or procedure is being used (for example, attempting to use a transformer that flips strategies on a rock-paper-scissors strategy)

Reply:
I'm fine with it if you want to leave it in but yeah as long as there's an explanation.

Reply:
it was a lot simpler than i thought to put together a little mock rock-paper-scissors with strategies & actions, and then run a tournament: https://gist.github.com/alexhroom/bea19c0bec62a08838bb6c8c4c949a6a

should I put this somewhere in the docs as a tutorial on adding different games to axelrod?

Reply:
A tutorial would be great. I just don't want someone to search for a library that can play RSP or NxN games for N > 2 and then be really disappointed to see that you can't really do it with axelrod. Like you technically can be not really since we only have IPD strategies so they'll only use an 2x2 submatrix.

More generally, and going back to the discussion on the issue about the library's abstractions, we probably need to add some way to filter strategies for compatibility with various games.

Reply:
Added a 'health warning' to the page on using different stage games and a full tutorial on implementing a new game with actions and strategies.